src/assignments/phat/lesson_5_answers.clj

```diff
@@ -0,0 +1,182 @@
+(ns src.assignments.phat.lesson-5-answers
+  (:require [clojure.java.io :as io]))
+
+;; write a unix-like grep command in clojure, so that, e.g.
+;; (grep "/tmp/somefile.txt" regex)
+;; ["match 1" "match 2" "..."]
+
+(def file "src/assignments/phat/tmp/somefile.txt")
+
+(def regex-patterns {:url "^https?://[^/]+/([^/]+)/.*$"
+                     :words "\\w+"
+                     :digit "\\d+"})
+
+(defn not-lazy-grep
+  "Returns a sequence of matches of pattern in a file"
+  [filepath regex]
+  (assert (.exists (io/file filepath))
+          "File path does not exist")
+  (let [re (re-pattern regex)]
+    (with-open [rdr (io/reader filepath)]
+      (re-seq re (slurp rdr)))))
+
+(not-lazy-grep file (:digit regex-patterns))
+
+(defn lazy-grep 
+  "Returns a lazy sequence of matches of a regex pattern from a file"
+  [filepath regex]
+  (assert (.exists (io/file filepath))
```

Comment:
Don't use assert for things you don't control.  I'll explain what I mean in class tonight.